packages/fast-components-react-base/src/foundation/foundation.spec.tsx

```diff
@@ -284,3 +284,147 @@ describe("generateClassNames", () => {
         ).toBe("component-classes custom-class-name");
     });
 });
+
+describe("withSlot", () => {
+    class WithSlotTestComponent extends Foundation<{}, {}, {}> {
+        public render(): React.ReactNode {
+            return this.props.children || "hello world";
+        }
+    }
+
+    test("should not throw if no props are passed", () => {
+        const Component: WithSlotTestComponent = new ReactTestUtils.renderIntoDocument(<WithSlotTestComponent />);
+
+        expect(() => {
+            Component["withSlot"]("test");
+        }).not.toThrow();
+    });
+
+    test("should return an empty array if slot no matches are found", () => {
```

Comment:
"should return an empty array if there are no children with a slot prop matching the provided slot are found"